### PR TITLE
Add release.{Lsb,ReadLsb()}

### PR DESCRIPTION
**Pull request by mvo5**
While working on the new classic-mode I noticed that I have to download a lxd image based on the lsb information we have. This branch adds support for reading lsb-release data .

When mvo5:refactor/less-cleanup-in-buildtest lands the tests can be somewhat simplified by using the AddCleanup helper.

### CODE REVIEW

**Comment by ogra1** [comment]
hmm, could that be switched to use /etc/os-release instead .... else we have another python dep in the rootfs (i would actually like to drop lsb-release from the image long term)

**Comment by mvo5** [comment]
@ogra1 I would love to, however /etc/os-release does not contain the code name in a sensible way (xenial) and that is what the lxd images are using.

File: release/release.go [hunk 1]
```diff
@@ -67,3 +75,37 @@ func Setup(rootDir string) error {
...
+	Codename string
+}
+
+// ReadLsb returns the lsb-release information of the current system
+func ReadLsb() (*Lsb, error) {
+	lsb := &Lsb{}
+
+	content, err := ioutil.ReadFile(lsbReleasePath)
+	if err != nil {
+		return nil, fmt.Errorf("can not read lsb-release: %s", err)
```
Comment (niemeyer) [inline release/release.go]: Nitpick: s/can not/cannot/

File: release/release_test.go [hunk 2]
```diff
@@ -50,3 +52,42 @@ func (s *ReleaseTestSuite) TestOverride(c *C) {
...
+`
+	err := ioutil.WriteFile(mockLsbRelease, []byte(s), 0644)
+	c.Assert(err, IsNil)
+
+	return mockLsbRelease
+}
+
+func (a *ReleaseTestSuite) TestReadLsb(c *C) {
+	realLsbRelease := release.LsbReleasePath()
+	defer func() { release.SetLsbReleasePath(realLsbRelease) }()
```
Comment (niemeyer) [inline release/release_test.go]: The comment in the addCleanup branch has a suggestion about a potential way to organize this logic which reduces the boilerplate in tests.

File: release/release_test.go [hunk 3]
```diff
@@ -50,3 +52,42 @@ func (s *ReleaseTestSuite) TestOverride(c *C) {
 	c.Check(release.String(), Equals, "10.06-personal")
 	c.Check(release.Get(), DeepEquals, rel)
 }
+
+func makeMockLsbRelease(c *C) string {
+	// FIXME: use AddCleanup here once available so that we
+	//        can do release.SetLsbReleasePath() here directly
```
Comment (niemeyer) [inline release/release_test.go]: Ah, interesting. That indeed makes me wish for something along the lines of addCleanup, as it's inside a helper which may be called or not, and the logic remains completely independent from the suite. I like it, thanks.

**Comment by niemeyer** [comment]
LGTM

File: release/release.go (hunk 1 above)
Comment (mvo5) [inline release/release.go]: Thanks! Fixed.

File: release/release_test.go (hunk 2 above)
Comment (mvo5) [inline release/release_test.go]: Thanks, excellent suggestion and a pattern I will keep in mind for the future! Fixed in git.

**Comment by ogra1** [comment]
i filed https://bugs.launchpad.net/snappy/+bug/1520154 to have /etc/os-release provide us the required field

**Comment by chipaca** [comment]
:+1:

**Pull request merged**